src/compositor/lipstickcompositorwindow.cpp

```diff
@@ -84,6 +84,11 @@ void LipstickCompositorWindow::setDelayRemove(bool delay)
         return;
 
     m_delayRemove = delay;
+    if (m_delayRemove)
+        disconnect(this, &QWaylandSurfaceItem::surfaceDestroyed, this, &QObject::deleteLater);
+    else
+        connect(this, &QWaylandSurfaceItem::surfaceDestroyed, this, &QObject::deleteLater);
```

Comment:
m_delayRemove is initialized as false, so you should do this connect in the ctor too.